frontend/src/pages/index.js

```diff
@@ -1,56 +1,28 @@
-import React, { useContext, useState, useEffect } from 'react'
-import { Card, Image, Placeholder } from 'semantic-ui-react'
+import React, { useContext } from 'react'
 import useSWR from 'swr'
 
-import styles from './index.module.scss'
-
 import { Page } from '@components/Page'
+import ProjectCard from '@components/ProjectCard'
 import { AuthContext } from '@contexts/AuthContext'
-import { getSession, getRepos } from '@utils/giteaApi'
-
-const processorUrl = process.env.KITSPACE_PROCESSOR_URL
+import { getAllRepos } from '@utils/giteaApi'
 
-const fetcher = (...args) => fetch(...args).then(r => r.json())
+export const getStaticProps = async () => {
+  const repos = await getAllRepos()
 
-const useThumbnail = full_name => {
-  const img = `/${full_name}/HEAD/images/top.png`
-  const statusUrl = processorUrl + '/status/' + img
-  const { data, error } = useSWR(statusUrl, fetcher, { refreshInterval: 1000 })
-  const isError = error || data?.status === 'failed'
   return {
-    src: processorUrl + '/files/' + img,
-    isLoading: !isError && data?.status !== 'done',
-    isError,
+    props: {
+      repos,
+    },
   }
 }
 
-const ProjectCard = ({ name, full_name, description, owner }) => {
-  const { src, isLoading, isError } = useThumbnail(full_name)
-  return (
-    <Card>
-      <div className={styles.thumbnail}>
-        <div>{isLoading || isError ? null : <img src={src} />}</div>
-      </div>
-      <Card.Content>
-        <Card.Header>{name}</Card.Header>
-        <Card.Meta>{owner.username}</Card.Meta>
-        <Card.Description>{description}</Card.Description>
-      </Card.Content>
-    </Card>
-  )
-}
-
 const Home = ({ repos }) => {
-  const { user, csrf } = useContext(AuthContext)
+  const { user } = useContext(AuthContext)
   const username = user?.login || 'unknown user'
-  const [projects, setProjects] = useState([])
 
-  useEffect(() => {
-    if (csrf) {
-      getRepos(csrf).then(setProjects)
-    }
-  }, [csrf])
+  const { data: projects } = useSWR('/', getAllRepos, { initialData: repos })
 
+  // TODO: handle the failure case i.e., projects is empty.
```

Comment:
You shouldn't have TODOs like this if you say it's ready to merge.

Reply:
I wasn't sure what to do in this case: it won't fail or raise any error, the `index` page won't have any projects just like this. Should the build deliberately fail in production in this case? display another component saying "Nothing here"? Or just leave it as is?

Reply:
Ah, I see. I thought you meant it crashes with no projects. Empty page is fine for now, given this shouldn't really happen. Once we add search it will be different again anyway.